manifests/bond/debian.pp

```diff
@@ -7,25 +7,24 @@
 # * Debian Network Bonding http://wiki.debian.org/Bonding
 define network::bond::debian(
   $slaves,
-  $ensure           = present,
-  $ipaddress        = undef,
-  $netmask          = undef,
-  $method           = undef,
-  $family           = undef,
-  $onboot           = undef,
-  $hotplug          = undef,
-  $mtu              = undef,
-  $options          = undef,
-  $slave_options    = undef,
-
-  $mode             = undef,
-  $miimon           = undef,
-  $downdelay        = undef,
-  $updelay          = undef,
-  $lacp_rate        = undef,
-  $primary          = undef,
-  $primary_reselect = undef,
-  $xmit_hash_policy = undef,
+  $ensure                           = present,
+  $ipaddress                        = undef,
+  $netmask                          = undef,
+  $method                           = undef,
+  $family                           = undef,
+  $onboot                           = undef,
+  $hotplug                          = undef,
+  Optional[Integer[42, 65536]] $mtu = undef,
```

Comment:
Rather than being backwards compatible, why not Variant[Integer[42,65536], String, Undef] and if $mtu =~ String { warning('$mtu should be an integer and will change in future releases') }